bcipy/gui/params_form.py

```diff
@@ -348,28 +373,91 @@ def onLoad(self, event: wx.EVT_BUTTON) -> None:
             if fd.ShowModal() == wx.ID_CANCEL:
                 return     # the user changed their mind
             load_file = fd.GetPath()
-            self.loaded_from.SetLabel(f'Loaded from: {load_file}')
-            self.flash_msg.SetLabel('Click the Save button to persist these '
-                                    'changes.')
-            self.vbox.Hide(self.form)
+
+            self.get_parameters_from_file(load_file)
+
+            self.params, missing_keys = check_if_parameters_contains_all_keys(self.params, load_file)
+
```

Comment:
The `check_if_parameters_contains_all_keys` name is a bit misleading, since this not only checks values but mutates the parameters dict. I think a better approach here would be to separate out this functionality into a few simpler functions that can be independently tested:

1. Get the missing key/values (diff) as a new dict:
https://stackoverflow.com/questions/32815640/how-to-get-the-difference-between-two-dictionaries-in-python
2. Merge the new diff dict into the params
3. Convert all keys to Parameter objects.